File: 1812.判断国际象棋棋盘中一个格子的颜色.cpp
/*
 * @lc app=leetcode.cn id=1812 lang=cpp
 *
 * [1812] 判断国际象棋棋盘中一个格子的颜色
 */

// @lc code=start
#include<string>
using namespace std;
class Solution {
public:
    bool squareIsWhite(string coordinates) {
        return ((coordinates[0]-'a') ^ (coordinates[1]-'1')) & 1;
    }
};
// @lc code=end

File: 1812.判断国际象棋棋盘中一个格子的颜色.py
#
# @lc app=leetcode.cn id=1812 lang=python3
#
# [1812] 判断国际象棋棋盘中一个格子的颜色
#

# @lc code=start
class Solution:
    def squareIsWhite(self, coordinates: str) -> bool:
        a=(ord(coordinates[0])-ord('a'))%2==0
        b=(int(coordinates[1])-1)%2==0
        return a!=b
# @lc code=end

print(Solution().squareIsWhite('c7'))